server/service/osquery_utils/queries.go

```diff
@@ -1428,8 +1428,8 @@ func directIngestMDMWindows(ctx context.Context, logger log.Logger, host *fleet.
 
 func directIngestMunkiInfo(ctx context.Context, logger log.Logger, host *fleet.Host, ds fleet.Datastore, rows []map[string]string) error {
 	if len(rows) == 0 {
-		// assume the extension is not there
-		return nil
+		// munki is not there, and we need to mark it deleted if it was there before
+		return ds.SetOrUpdateMunkiInfo(ctx, host.ID, "", []string{}, []string{})
```

Comment:
The query `select <stuff> from munki_info` is part of detail query which is hardcoded in fleetd on the host. If Munki doesn't exist (or was completely wiped) on the host, the query will return no rows. The server will see that as it is processing detail query results from the host.